hazelcast-jet-distribution/src/root/config/hazelcast.yaml

```diff
@@ -1,30 +1,78 @@
-# You may obtain a copy of the License at
-#
-# http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
 hazelcast:
+  # The name of the cluster and must match what's provided by clients
+  # for connection and the other nodes in the cluster
   cluster-name: jet
   network:
+    port:
+      # The port number the server should start searching for ports. If
+      # the port is taken Jet will continue searching for additional ports
+      # starting from the given port.
+      port: 5701
+      auto-increment: true
+    # Which network interface the server should listen on. If enabled is set
+    # to false, it will listen on all available interfaces
+    interfaces:
+      enabled: false
+      interfaces:
+        - 127.0.0.1
+    # The join section configures the strategies used by the server to
+    # discover other Jet nodes. Jet will form a cluster automatically with
+    # the other nodes using the same discovery mechanism. Only one join
+    # config should be active at a time, and it should be shared by all
+    # the nodes
     join:
+      # The default way to discover nodes is using multicast, when multicast
+      # is not enabled for the network, it is recommended that you use tcp-ip
+      # discovery instead.
       multicast:
         enabled: true
         multicast-group: 224.2.2.3
         multicast-port: 54328
+      # Enable this to explicitly specify the IPs for the other nodes in the cluster
       tcp-ip:
         enabled: false
         member-list:
-          - 127.0.0.1
-    port:
-      auto-increment: true
-      port-count: 100
-      port: 5701
+          - 10.0.0.1
+          - 10.0.0.2
+      # You can use this strategy for dynamic discovery on AWS without
+      # having to configure the IPs explicitly. Instances matching the
+      # given tag key and value will be automatically joined to. For more
+      # details see https://github.com/hazelcast/hazelcast-aws
+      aws:
+        enabled: false
+        access-key: my-access-key
+        secret-key: my-secret-key
+        iam-role: dummy
+        region: us-west-1
+        host-header: ec2.amazonaws.com
+        security-group-name: hazelcast-sg
+        tag-key: type
+        tag-value: hz-nodes
+      # For details on how to configure discovery on GCP please see
+      # https://github.com/hazelcast/hazelcast-gcp
+      gcp:
+        enabled: false
+        label: application=hazelcast-jet
+        zones: us-east1-b,us-east1-c
+    # Allows access to cluster via the REST API
+    rest-api:
+      enabled: true
+      endpoint-groups:
+        CLUSTER_READ:
+          enabled: true
+        CLUSTER_WRITE:
+          enabled: true
```

Comment:
this enables cluster shutdown and it is `disabled` by default for IMDG, is it a good idea to enable it by default ?

Reply:
It is useful for doing shutdown from MC and using the cluster.sh scripts. I thought we enable it by default so those scripts work.